javaparser-core/src/main/java/com/github/javaparser/ast/StubUnit.java

```diff
@@ -0,0 +1,153 @@
+package com.github.javaparser.ast;
+
+import com.github.javaparser.TokenRange;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+import com.github.javaparser.metamodel.InternalProperty;
+import com.github.javaparser.printer.PrettyPrinter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.github.javaparser.Providers.UTF8;
+
+/**
+ * <p>
+ * This class represents a stub file. The stub file is a Java file, but with the optionally omitted
+ * information that is not relevant to pluggable type-checking; this makes the stub file smaller
+ * and easier for people to read and write.
+ * </p>
+ * A stub unit contains the list of compilation units.
+ * This class copied the {@link CompilationUnit} and then adjusted to the needs of the Checker Framework.
+ *
+ * @see CompilationUnit
+ */
+public class StubUnit extends Node {
+
+    /**
+     * The field represents a list of compilations units.
+     */
+    private NodeList<CompilationUnit> compilationUnits;
+
+    /** Contains the information about where this stub unit was loaded from, or empty if it wasn't loaded from a file.*/
+    @InternalProperty
+    private StubUnit.Storage storage;
+
+    /**
+     * The constructor that takes the tokenRange and just pass it to the super method {@link Node}.
```

Comment:
Internal implementation details don't belong in Javadoc.  It's better to say what the TokenRange is or represents or how it is used, if possible.

Also, Javadoc should say what the method does, not what it is.  Rather than, "the constructor that...", write something like "Constructs a StubUnit from a list of tokens".